### ISSUE TITLE
False positive "space-before-keywords" for function in array

### ISSUE THREAD

**Issue by spence** [eslint/eslint#4569]
Spaces before keyword `function` when it's the first item in an array should not cause an error.

```javascript
'use strict';
(function () {
  window._ = [function () { }];
})();
```

```
3:15  error  Missing space before keyword "function"  space-before-keywords
```

eslint version: v1.10.2

Rule:
```
"space-before-keywords": [2, "always"]
```

**Comment by eslintbot** [comment]
@spence Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by platinumazure** [comment]
Yes, this is a conflict with array-bracket-spacing and I would agree this
rule should not be enforced in that situation. I think the colon space rule
will also be problematic in the object case (can someone check that?).

**Comment by alberto** [comment]
Should this be tagged `enhancement` @eslint/eslint-team?

**Comment by nzakas** [comment]
No, I think bug is correct
